internals/src/macros.rs

```diff
@@ -2,6 +2,8 @@
 //!
 
 /// Implements standard array methods for a given wrapper type.
+// Types that use this macro should derive `Copy` and `Clone` so that the provided conversion
+// methods are named according to Rust convention.
```

Comment:
We could statically check this:

```rust
fn check_copy<T: Copy>() {}
check_copy::<$ty>();
```

Reply:
Nice, I didn't think of that :)  Will add.

Reply:
I couldn't work out exactly where you meant to call `check_copy`, I came up with this:
```rust

            /// This function is not meant to be called. Checks at compile time that `$thing`
            /// implements `Copy`.
            fn _check_copy(self) {
                fn _check<T: Copy>() {}
                _check::<$thing>();
            }
```

Reply:
I meant to write both declaration and call inside `to_bytes()`.